packages/client/src/comp/states/Base.tsx

```diff
@@ -23,75 +27,154 @@ type Props<Info> = React.PropsWithChildren<{
-      name: nameRef.value() || '',
-      birthdate: birthdateRef.value() || '',
-      email: emailRef.value() || '',
-      mailingAddress: mailingRef.value() || '',
-      phone: phoneRef.value() || '',
+      name: `${fields.firstName.value} ${fields.lastName.value}`,
+      birthdate: fields.birthdate.value,
+      email: fields.email.value,
+      mailingAddress: fields.mailing.value,
+      phone: fields.telephone.value,
       uspsAddress,
       contact,
       address,
     }
 
-    const info = enrichValues(baseInfo)
+    const info = enrichValues(base)
+
     if (!info) {
       // Do not dismiss previous errors which may give more details on bad fields
       toast.error('Please fill all the required fields in the right formats')
-      return
     }
+
+    return info
+  }
+
+  async function checkRegistration(
+    id: 'firstName' | 'lastName' | 'birthdate',
+    e: React.ChangeEvent<HTMLInputElement>,
+  ) {
+    e.preventDefault()
+    e.persist() // allow async function call
+    updateField(id, e.currentTarget.value)
+    if (canCheckRegistration()) {
+      if (registrationStatus !== 'Ignored' && registrationStatus !== 'Active') {
+        const {
+          firstName, lastName, birthdate,
+        } = fields
+        setRegistrationStatus('Loading')
+
+        const result = await client.isRegistered({
+          firstName: firstName.value,
+          lastName: lastName.value,
+          birthdate: birthdate.value,
+          city: address?.city ?? '',
+          postcode: address?.postcode ?? '',
+          stateAbbr: address?.stateAbbr ?? '',
+          street: address?.street ?? '',
+          streetNumber: address?.streetNumber ?? '',
+        })
+        if (result.type === 'data') {
+          setRegistrationStatus(result.data ?? 'Error')
+        } else {
+          setRegistrationStatus('Error')
+        }
+      }
+    }
+  }
+
+  // event is optional so we can use this function outside the form, i.e. modals
+  async function handleSubmit(event?: React.FormEvent<HTMLFormElement>) {
+    event?.persist()  // allow async function call
+    event?.preventDefault()
+
+    const info = stateInfo()
+    if (!info) return
+
     setFetchingData(true)
     const result = await client.register(info, voter)
-    setFetchingData(false)
     if(result.type === 'data'){
       pushSuccess(result.data)
     } else {
       toast.error('Error signing up.  Try resubmitting.  If this persists, try again in a little while.')
     }
+    setFetchingData(false)
   }
 
   return <AppForm onSubmit={handleSubmit}>
-    <NameInput
-      id='name'
-      ref={nameRef}
-      defaultValue={query.name}
-      required
-    />
+    <NameWrapper>
+      <NameInput
+        id='firstName'
+        value={fields.firstName.value}
+        label='First Name'
+        defaultValue={query.name}
+        invalid={!fields.firstName}
+        onChange={e => {
+          // Reset registration status on name change
+          setRegistrationStatus(null)
```

Comment:
I noticed that the status box disappears whenever you type a letter.  Can we get rid of this reset (don't know why it's here).